Add psp-gdb instruction (#34)
pspdev/psplinkusb: README.md

If no result is returned, make sure to build with the ``-g`` or``-g3`` option to make sure psp-addr2line knowns the function names and locations.

## Debugging with GDB

Sometimes debugging with the built in instruction level debugger is not enough, this is where
PSPLINK's GDB Server can come in handy.

Let's get started:

### Preparation

> Prepare a separate terminal for `usbhostfs_pc`, `pspsh` and `psp-gdb`. All of them are needed to point the directory location to your `.prx` and `.elf` files located or the file you need to debug.

Reviewer: ```suggestion
Prepare a separate terminal for `usbhostfs_pc`, `pspsh` and `psp-gdb`. Open all of them in the directory in which your compiled `.prx` and `.elf` files are located.
```

Reviewer: I'm not sure if this suggestion makes sense, since pspsh does noy need to be opened in a specific directory.

### 1. usbhostfs_pc

> Run `usbhostfs_pc` on your terminal dedicated for `usbhostfs_pc` and you will see the `waiting for device...` status and then start the PSPLINK app on your PSP and connect the USB cable, you will see the `connected to device` status, it means success. do not close this terminal after that.

Reviewer: ```suggestion
Run `usbhostfs_pc` on your terminal dedicated for `usbhostfs_pc` and you will see the `waiting for device...` status. Now start the PSPLINK app on your PSP and connect the USB cable. You should see the `connected to device` status in the terminal, which means success. Do not close this terminal after that.
```

Reviewer: Not sure if this suggestion is the best wording, but the long sentence would be nice to break up.

### 2. pspsh

> Run `pspsh` on your terminal dedicated for `pspsh` and you will see the `host0:/>`. After that you need to change again the directory where your files needed to debug. lastly, run `debug <file.prx>` and it will display something like this:

Reviewer: You probably won't need to change directory. It will open where you ran usbhostfs_pc

Reviewer: I would remove <> from the debug command for clarity

> You need to replace `<file.prx>`  with the file you need to debug. PRX file not ELF.

```sh
PSPLink USB GDBServer (c) 2k7 TyRaNiD
host0:/> Loaded host0:/<file.prx> - UID 0x0408A763, Entry 0x088040AC
```

It means the debuggee is succesfully loaded. You can type `reset` if there's something wrong with your GDBServer.

### 3. psp-gdb

> Run `psp-gdb <file.elf> -q` on your terminal dedicated for `psp-gdb` and you will see something like this:

Reviewer: The elf file has no extension, but has the same name as the prx otherwise. Might want to clarify that.
